src/request_parser.rs

```diff
@@ -0,0 +1,16 @@
+//! Methods for retrieving swagger-related information from an HTTP request.
+use hyper::Request;
+
+/// A trait for retrieving swagger-related information from a request.
+///
+/// This allows other middlewares to retrieve API-related information from a request that
+/// may not have been handled by the autogenerated API code yet.   For example, a statistics
+/// tracking service may wish to use this to count requests per-operation.
+///
+/// The trait is automatically implemented by swagger-codegen.
+pub trait RequestParser {
+    /// Retrieve the Swagger operation identifier that matches this request.
+    ///
+    /// Returns `Err(())` if this request does not match any known operation on this API.
+    fn get_operation(req: &Request) -> Result<&'static str, ()>;
```

Comment:
Should this be `get_operation_id`?  This gives flexibility if we need to return a parsed operation struct in future.

Reply:
Works for me.  In fact, is it more Rusty to drop the `get` and just call it `operation_id`?